src/components/debug-modal/debug-modal.jsx

```diff
@@ -0,0 +1,174 @@
+import React, {useState, useCallback, useEffect} from 'react';
+import {defineMessages, FormattedMessage} from 'react-intl';
+import PropTypes from 'prop-types';
+import ReactModal from 'react-modal';
+import classNames from 'classnames';
+import {sections} from './sections/sections';
+import GA4 from '../../lib/analytics';
+
+import styles from './debug-modal.css';
+import debugIcon from './icons/icon--debug.svg';
+import debugIconInverted from './icons/icon--debug-inverted.svg';
+import closeIcon from './icons/icon--close.svg';
+import prevIcon from './icons/icon--prev.svg';
+import nextIcon from './icons/icon--next.svg';
+
+const messages = defineMessages({
+    title: {
+        id: 'gui.debugModal.title',
+        defaultMessage: 'Debugging | Getting Unstuck',
+        description: 'title for the debugging modal'
+    }
+});
+
+const logTopicChange = topicIndex => {
+    GA4.event({
+        category: 'change_topic_debug_modal',
+        label: sections[topicIndex].id
+    });
+};
+
+const DebugModal = ({isOpen, onClose = () => {}}) => {
+    const [selectedTopicIndex, setSelectedTopicIndex] = useState(0);
+
+    // Preload images
+    useEffect(() => {
+        sections.forEach(section => {
+            new Image().src = section.image;
+        });
+    }, []);
```

Comment:
The idea is, that by loading the image once, React caches it and can reuse it later. So the earlier we execute this, the more time we have to load all images we need - in this case, on first render.
I added this at a relatively late point in developing the debug modal since I noticed an issue with the image load time. There seemed to be a noticeable improvement after that (though I did not measure it). Interestingly, I learned about this pattern from Scratch - I noticed it is used in a few places in `scratch-www`